bin/dev

```diff
@@ -63,135 +74,143 @@ runWebpack(){
-    echo $cmd
-    exec $cmd
+    runDockerExec "pytest "$*
 }
 
-# Run migrate
+# Run migration script
 runMigrate(){
-    findBackendContainerId container_id
-    cmd="docker exec -it "$container_id" pipenv run python3 -m src.db.runner"
-    echo $cmd
-    exec $cmd
+    runDockerExec "pipenv run python3 -m src.db.runner"
 }
 
 # Run js tests
+# args: list of args to pass to npm run test
 runJstest(){
-    cd client
-    npm run test $*
+    bin/util logAndRun "cd client"
+    bin/util logAndRun "npm run test $*"
 }
 
 # Run precommit inside container
+# args: list of args to pass to pre-commit
 runPrecommit(){
-    findBackendContainerId container_id
     # if no cli args, run standard pre-push
     if [ ! $* ]; then
-    args="run --hook-stage push --from-ref origin/main --to-ref HEAD"
+        args="run --hook-stage push --from-ref origin/main --to-ref HEAD"
     else
-    args=$*
+        args=$*
     fi
-    cmd="docker exec -t -w /home/worker/app/ "$container_id" pre-commit "$args
-    echo $cmd
-    exec $cmd
+    runDockerExecNonInteractive "pre-commit "$args
 }
 
-# One time setup to add pre-commit hook
+# One time setup to change git hooks path
 runSetup(){
-    cp bin/pre-commit .git/hooks/pre-push
+    bin/util logAndRun "git config core.hooksPath bin"
 }
 
-printHelp(){
-echo "
-SyncM8 command runner
-
-
-bin/dev COMMAND [options]
-
-
-
-backend
-    Create containers if not up and attach api container to shell to faciliatate pdb.
-
-build
-    Build images for all syncm8 containers.
-
-check
-    Run pre-commit inside api container. All [options] are transparently passed.
-
-clean
-    Stop and remove all active containers and delete all db data associated with syncm8.
-
-debug
-    Alias for backend.
-
-down
-    Stop and remove all active containers associated with syncm8.
-
-frontend
-    Run frontend webpack and attach shell.
-
-jstest
-    Run js tests. All [options] are transparently passed.
-
-migrate
-    Run migration scripts for database.
-
-pre-commit
-    Run pre-commit inside api container. All [options] are transparently passed.
-
-ps
-    Show running docker images associated with syncm8.
-
-pytest
-    Run pytest inside the api container. All [options] are transparently passed.
-
-rebuild
-    Force rebuild all containers from scratch and update to newest versions.
-    N.b. All db data will be lost. This is primarily meant as last resort.
-
-setup
-    One time setup to add pre-commit hook to git repo.
+# Pull images from aws ECR
+runPull(){
+    bin/util logAndRun "bin/infra pull"
+}
 
-webpack
-    Alias for frontend.
-"
+# Hacky solution required for host machine to use python inside venv for linting python
+# Remove the symlink to py3, copy the whole py3 binary, fix the other symlinks to point to this binary
+runFixPythonLoc(){
+    runDockerExec "rm /home/worker/app/server/.venv/bin/python3.8"
+    runDockerExec "cp /usr/local/bin/python3.8 /home/worker/app/server/.venv/bin/python3.8"
```

Comment:
lol how long does this take to cp py3 binary?

Reply:
basically zero time